src/paste-markdown-html.ts

```diff
@@ -20,65 +18,80 @@ function onPaste(event: ClipboardEvent) {
   if (!(field instanceof HTMLTextAreaElement)) return
 
   // Get the plaintext and html version of clipboard contents
-  let text = transfer.getData('text/plain')
+  let plaintext = transfer.getData('text/plain')
   const textHTML = transfer.getData('text/html')
   // Replace Unicode equivalent of "&nbsp" with a space
-  const textHTMLClean = textHTML.replace(/\u00A0/g, ' ')
+  const textHTMLClean = textHTML.replace(/\u00A0/g, ' ').replace(/\uC2A0/g, ' ')
   if (!textHTML) return
 
-  text = text.trim()
-  if (!text) return
+  plaintext = plaintext.trim()
+  if (!plaintext) return
 
   // Generate DOM tree from HTML string
   const parser = new DOMParser()
   const doc = parser.parseFromString(textHTMLClean, 'text/html')
+  const walker = doc.createTreeWalker(doc.body, NodeFilter.SHOW_ELEMENT)
 
-  const a = doc.getElementsByTagName('a')
-  const markdown = transform(a, text, linkify as MarkdownTransformer)
+  const markdown = convertToMarkdown(plaintext, walker)
 
   // If no changes made by transforming
-  if (markdown === text) return
+  if (markdown === plaintext) return
 
   event.stopPropagation()
   event.preventDefault()
 
   insertText(field, markdown)
 }
 
-// Build a markdown string from a DOM tree and plaintext
-function transform(
-  elements: HTMLCollectionOf<HTMLElement>,
-  text: string,
-  transformer: MarkdownTransformer,
-  ...args: string[]
-): string {
-  const markdownParts = []
-  for (const element of elements) {
-    const textContent = element.textContent || ''
-    const {part, index} = trimAfter(text, textContent)
-    if (index >= 0) {
-      markdownParts.push(part.replace(textContent, transformer(element, args)))
-      text = text.slice(index)
+function convertToMarkdown(plaintext: string, walker: TreeWalker): string {
+  let currentNode = walker.firstChild() as HTMLAnchorElement | HTMLElement | null
+  let markdown = plaintext
+  let markdownIgnoreBeforeIndex = 0
+  let index = 0
+  const NODE_LIMIT = 100000
```

Comment:
This might be a bit high. I imagine it'll take over 100ms to walk this many nodes. Seems unlikely we'll experience people pasting 100k nodes?

Reply:
That's fair, let's reduce it to 10k! I figure hitting an upper bound like this is highly unlikely and should correlate with the [maximum issue/PR size](https://github.community/t/maximum-length-for-the-comment-body-in-issues-and-pr/148867). I could imagine complicated DOM trees that would exceed 1k nodes, so 10k seems like a reasonable cap.

Playground [here](https://codepen.io/imjohnbo/pen/dyJreaj) if interested.